macros/src/memoize.rs

```diff
@@ -118,14 +124,31 @@ fn process(function: &Function) -> Result<TokenStream> {
         ident.mutability = None;
     }
 
-    let unique = quote! { __ComemoUnique };
     wrapped.block = parse_quote! { {
-        struct #unique;
+        static __CACHE: ::comemo::internal::Lazy<
```

Comment:
Perhaps the whole `Lazy<RwLock<Cache<Args<...>, ..>>>` should be abstracted away behind a const-constructable type in the non-macro code? The `Cache` could manage the `Lazy<RwLock<..>>` and the lazy registration itself and the actual data is stored in something like `CacheData`. Then, we'd have less macro-generated code and less `comemo::internal` exports.